crossbeam-channel/src/channel.rs

```diff
@@ -656,14 +656,52 @@ impl<T> Sender<T> {
             _ => false,
         }
     }
+
+    /// Disconnects the channel.
+    ///
+    /// Explicitly disconnects the channel and returns `true`, as if all instances of either side
+    /// (sender or receiver) are dropped, unless it has been already disconnected. Otherwise, this
+    /// method does nothing and returns `false`.
+    ///
+    /// The successful disconnect operation results in immediately rejecting sending subsequent
+    /// messages to receivers.
+    ///
+    /// Disconnected channels can be restored connected by calling [`connect`], as long as there
+    /// are at least a sender and a receiver for the channel.
+    ///
+    /// [`connect`]: Self::connect
+    pub fn disconnect(&self) -> bool {
+        match &self.flavor {
+            SenderFlavor::List(chan) => chan.disconnect(),
+            _ => todo!(),
```

Comment:
obviously, i need to add support other flavors. this is true for other 3 `todo!()`s.